pkg/syncer/translator/from_config_translator.go

```diff
@@ -0,0 +1,177 @@
+package translator
+
+import (
+	"strings"
+
+	"github.com/loft-sh/vcluster/pkg/constants"
+
+	"github.com/loft-sh/vcluster/pkg/syncer/synccontext"
+	syncer "github.com/loft-sh/vcluster/pkg/syncer/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type fromHostTranslate struct {
+	gvk            schema.GroupVersionKind
+	eventRecorder  record.EventRecorder
+	virtualToHost  map[string]string
+	hostToVirtual  map[string]string
+	namespace      string
+	translatorName string
+	skipFuncs      []ShouldSkipHostObjectFunc
+}
+
+// ShouldSkipHostObjectFunc takes object's host name and namespace
+// and returns true if object should be skipped in the from host sync.
+type ShouldSkipHostObjectFunc func(hostName, hostNamespace string) bool
+
+func NewFromHostTranslatorForGVK(ctx *synccontext.RegisterContext, gvk schema.GroupVersionKind, hostToVirtual map[string]string, skipFuncs ...ShouldSkipHostObjectFunc) (syncer.FromConfigTranslator, error) {
+	virtualToHost := make(map[string]string, len(hostToVirtual))
+	for host, virtual := range hostToVirtual {
+		virtualToHost[virtual] = host
+	}
+
+	return &fromHostTranslate{
+		gvk:            gvk,
+		eventRecorder:  ctx.VirtualManager.GetEventRecorderFor("from-host-" + strings.ToLower(gvk.Kind) + "-syncer"),
+		virtualToHost:  virtualToHost,
+		hostToVirtual:  hostToVirtual,
+		namespace:      ctx.Config.ControlPlaneNamespace,
+		translatorName: "from-host-" + strings.ToLower(gvk.Kind),
+		skipFuncs:      skipFuncs,
+	}, nil
+}
+
+func (c *fromHostTranslate) Name() string {
+	return c.translatorName
+}
+
+func (c *fromHostTranslate) Resource() client.Object {
+	switch c.gvk.Kind {
+	case "ConfigMap":
+		return &corev1.ConfigMap{}
+	case "Secret":
+		return &corev1.Secret{}
+	default:
+		return nil
```

Comment:
Asking because I don't know.  Would it ever be ok to return a nil here?  Its rarely the case that a panic is the right thing to do, but this _**might**_ be one case.  Thoughts?

Reply:
It's only for now. Once this is merged, I'll open another PR to re-use this translator for CRDs - and I already changed this default case there.